src/instrumenter/instrumenter.ts

```diff
@@ -15,36 +15,18 @@ ts.sys.startReplay("${ logFilename }");`);
 
 function instrument(tscPath: string, prepareCode: string, cleanupCode = "") {
     const bak = `${tscPath}.bak`;
-    fs.exists(bak, (backupExists: boolean) => {
-        let filename = tscPath;
-        if (backupExists) {
-            filename = bak;
-        }
-
-        fs.readFile(filename, "utf-8", (err: any, tscContent: string) => {
-            if (err) throw err;
-
-            fs.writeFile(bak, tscContent, (err: any) => {
-                if (err) throw err;
-
-                fs.readFile(path.resolve(path.dirname(tscPath) + "/loggedIO.js"), "utf-8", (err: any, loggerContent: string) => {
-                    if (err) throw err;
-
-                    const invocationLine = "ts.executeCommandLine(ts.sys.args);";
-                    const index1 = tscContent.indexOf(invocationLine);
-                    if (index1 < 0) {
-                        throw new Error(`Could not find ${invocationLine}`);
-                    }
-
-                    const index2 = index1 + invocationLine.length;
-                    const newContent = tscContent.substr(0, index1) + loggerContent + prepareCode + invocationLine + cleanupCode + tscContent.substr(index2) + "\r\n";
-                    fs.writeFile(tscPath, newContent, err => {
-                        if (err) throw err;
-                    });
-                });
-            });
-        });
-    });
+    const filename = fs.existsSync(bak) ? bak : tscPath;
+    const tscContent = fs.readFileSync(filename, "utf-8");
+    fs.writeFileSync(bak, tscContent);
+    const loggerContent = fs.readFileSync(path.resolve(path.dirname(tscPath) + "/loggedIO.js"), "utf-8")
+    const invocationLine = "ts.executeCommandLine(ts.sys, ts.noop, ts.sys.args);";

```

Comment:
the text to search for has changed slightly, and is the only real change in this file